src/coreclr/src/jit/importer.cpp

```diff
@@ -4226,6 +4226,39 @@ GenTree* Compiler::impMathIntrinsic(CORINFO_METHOD_HANDLE method,
 
     op1 = nullptr;
 
+    if (intrinsicID == CORINFO_INTRINSIC_Pow && impStackTop().val->IsCnsFltOrDbl())
```

Comment:
Are there cases we are going to miss by handling this in the importer, rather than later in morph or lowering?

Reply:
@tannergooding not sure honestly, but "inlining" cases are handled, e.g.
```csharp
float Foo(float x, float y) => MathF.Pow(x, y);

float Test(float x) => Foo(x, 2.0f);
```
`Test` will be optimized.

Reply:
However, not sure about this one:
```csharp
float Foo(float x, float y) => MathF.Pow(x, y + 1.0f);
float Test(float x) => Foo(x, 1.0f);
```
(constant folding after inlining)

Reply:
Consider adding a unit test for that case.

Reply:
Consider adding a unit test for that case.

Reply:
Added a test and yes, the `y + 1.0` is handled fine (importer will fold 1+1) during inlining.